exec/java-exec/src/test/java/org/apache/drill/test/TestGracefulShutdown.java

```diff
@@ -201,24 +201,21 @@ public void testRestApi() throws Exception {
     builder = enableWebServer(builder);
     QueryBuilder.QuerySummaryFuture listener;
     final String sql = "select * from dfs.root.`.`";
-    try ( ClusterFixture cluster = builder.build();
+    try (ClusterFixture cluster = builder.build();
           final ClientFixture client = cluster.clientFixture()) {
       Drillbit drillbit = cluster.drillbit("db1");
-      int port = drillbit.getContext().getConfig().getInt("drill.exec.http.port");
-      int grace_period = drillbit.getContext().getConfig().getInt(ExecConstants.GRACE_PERIOD);
+      int port = drillbit.getWebServerPort();
+      int gracePeriod = drillbit.getContext().getConfig().getInt(ExecConstants.GRACE_PERIOD);
       listener =  client.queryBuilder().sql(sql).futureSummary();
       Thread.sleep(60000);
-      while( port < 8049) {
-        URL url = new URL("http://localhost:"+port+"/gracefulShutdown");
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-        conn.setRequestMethod("POST");
-        if (conn.getResponseCode() != 200) {
-          throw new RuntimeException("Failed : HTTP error code : "
-                  + conn.getResponseCode());
-        }
-        port++;
+      URL url = new URL("http://localhost:" + port + "/gracefulShutdown");
+      HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+      conn.setRequestMethod("POST");
+      if (conn.getResponseCode() != 200) {
+        throw new RuntimeException("Failed : HTTP error code : "
+                + conn.getResponseCode());
       }
-      Thread.sleep(grace_period);
+      Thread.sleep(gracePeriod);
```

Comment:
I don't think we need this sleep here. We can change the loop below to check for the termination condition.

```
while (currentTime < stopTime) {
  Collection<DrillbitEndpoint> drillbitEndpoints = cluster.drillbit()
               .getContext()
               .getClusterCoordinator()
               .getOnlineEndPoints();
  if (drillbitEndpoints.size() == 2) {
    break;
  }

  Thread.sleep(100L);
  currentTime = System.currentTimeMillis();
}
```

Reply:
In fact since we are doing the same thing in the other test could you move this code into a helper method?